### PR TITLE
Improvements & feedback

### PR DESCRIPTION
**Pull request by ElviaBth**
🎩 What? Why?
Add improvements 
- [x] On a survey component's admin page, there is a link “Create answer codes here”: The should open in a new tab `target="_blank"`
- [x] In the same page, also open in a new tab if there's a code applied that takes you to the survey
- [x] Add a link to administrate the component survey from the anonymous codes groups page
- [x] Allow admins to introduce manual codes instead of automatics

📌 Related Issues
#20 

📷 Screenshots
image
code_group index
image
create manual tokens view

♥️ Thank you!

### CODE REVIEW

File: app/views/decidim/anonymous_codes/admin/code_groups/index.html.erb [hunk 1]
```diff
@@ -34,6 +34,7 @@
                   <%= icon_link_to "list", code_group_codes_path(group), t("actions.list_tokens", scope: "decidim.anonymous_codes.admin"), class: "action-icon--preview#{allowed_to?(:view, :anonymous_code_token) ? '':' invisible'}" %>
                   <%= icon_link_to "pencil", edit_code_group_path(group), t("actions.edit", scope: "decidim.anonymous_codes.admin"), class: "action-icon--edit#{allowed_to?(:update, :anonymous_code_group, code_group: group) ? '' : ' invisible'}" %>
                   <%= icon_link_to "circle-x", code_group_path(group), t("actions.destroy", scope: "decidim.anonymous_codes.admin"), method: :delete, class: "action-icon--remove#{allowed_to?(:destroy, :anonymous_code_group, code_group: group) ? '' : ' invisible'}", data: { confirm: t("actions.confirm_destroy", scope: "decidim.anonymous_codes.admin") } %>
+                  <%= icon_link_to "cog", surveys.first[:edit_path], t("actions.survey", scope: "decidim.anonymous_codes.admin"), class: "action-icon--edit#{group.tokens_count > 0 ? '' : ' invisible'}", target: "_blank" %>
```
Comment (microstudi) [inline app/views/decidim/anonymous_codes/admin/code_groups/index.html.erb]:
```suggestion
                  <%= icon_link_to "cog", edit_component_path(group.resource), t("actions.survey", scope: "decidim.anonymous_codes.admin"), class: "action-icon--edit#{group.resource.present? ? '' : ' invisible'}", target: "_blank" %>
```

File: app/controllers/decidim/anonymous_codes/admin/code_groups_controller.rb [hunk 2]
```diff
@@ -100,10 +100,12 @@ def surveys
...
-              [
-                "#{translated_attribute(component.participatory_space.title)} :: #{translated_attribute(component.name)}",
-                survey.id
-              ]
+              edit_path = Decidim::EngineRouter.admin_proxy(component.participatory_space).edit_component_path(component.id)
+              {
+                title: "#{translated_attribute(component.participatory_space.title)} :: #{translated_attribute(component.name)}",
+                survey_id: survey.id,
+                edit_path: edit_path
+              }
```
Comment (microstudi) [inline app/controllers/decidim/anonymous_codes/admin/code_groups_controller.rb]:
```suggestion
              }
              end
              
              def edit_component_path(resource)
                return unless resource
                Decidim::EngineRouter.admin_proxy(resource.component.participatory_space).edit_component_path(resource.component.id)
              end
```

Comment (ElviaBth) [reply app/controllers/decidim/anonymous_codes/admin/code_groups_controller.rb]: Done

File: app/views/decidim/anonymous_codes/admin/code_groups/index.html.erb (hunk 1 above)
Comment (ElviaBth) [reply app/views/decidim/anonymous_codes/admin/code_groups/index.html.erb]: Done

File: app/commands/decidim/anonymous_codes/admin/create_tokens.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          @form = form
+          @code_group = code_group
+        end
+
+        def call
+          return broadcast(:invalid) if form.invalid?
+
+          transaction do
+            create_code_group!
+          end
```
Comment (microstudi) [inline app/commands/decidim/anonymous_codes/admin/create_tokens.rb]: As this SQL is already atomic, there's no need to specify a transaction here, so you can simply
```suggestion
          Token.create!(group: code_group, token: form.token)
```

**Review by microstudi — COMMENTED** [review]
Nice job, I've left you a comment about a transaction.
Also, I think that we should call the class `CreateTokens` just `CreateToken`, in singular as it creates just one token.

Then, we need a system spec for the creation token step. 

And finally, see if you can add to the README this text in the table explaining the ENV vars:

```
| ANONYMOUS_CODES_MANUAL_TOKEN_REGEX | The regular expression to validate manual tokens (by default allows only basic letters and numbers, always uppercase). | `\A[A-Z0-9]*\z` |
```

File: app/commands/decidim/anonymous_codes/admin/create_tokens.rb (hunk 3 above)
Comment (ElviaBth) [reply app/commands/decidim/anonymous_codes/admin/create_tokens.rb]: Thanks

**Review by microstudi — APPROVED** [review]
nice job!

**Pull request merged**